doc/specs/stdlib_bitsets.md

```diff
@@ -0,0 +1,2006 @@
+
+`status` (optional): shall be a scalar default integer variable. It is
+an `intent(out)` argument that, if present, returns an error code
+indicating any problem found in processing `init`, and if absent and
+an error was found result in stopping processing with an informative
+stop code. It can have any of the following error codes:
+
+* `success` - no problem found
+
+* `alloc_fault` - `self` was of type `bitset_large` and memory
+  allocation failed
+
+* `array_size_invalid_error` - bits was present with either a negative
+  value, or a value greater than 64 when `self` was of type
+  `bitset_64`.
+
+#### Example
+
+```fortran
+    program demo_init
+        use stdlib_bitsets
+        type(bitset_large) :: set0
+        call set0 % init(166)
+        if ( set0 % bits() == 166 ) &
+            write(*,*) `SET0 has the proper size.'
+        if ( set0 % none() ) write(*,*) 'SET0 is properly initialized.'
+    end program demo_init
+```
+
+### `input` - reads a bitset from an unformatted file
+
+#### Status
+
+Experimental
+
+#### Description
+
+Reads a bitset from its binary representation in an unformatted
+file.
+
+#### Syntax
+
+`call self % [[bitset_type(class):input(bound)]] (unit [, status])`
+
+#### Class
+
+Subroutine
+
+#### Arguments
+
+`self`: shall be a scalar variable of class `bitset_64` or
+`bitset_large`. It is an `intent(out)` argument.
+
+`unit`: shall be a scalar default integer expression. It is an
+`intent(in)` argument. Its value must be that of a logical unit
+number for an open unformatted file with `READ` or `READWRITE`
+access positioned at the start of a BITSET value written by a
+`bitset_type` `output` subroutine by the same processor.
+
+`status` (optional): shall be a scalar default integer variable. If
+present its value shall be of one of the error codes defined in this
+module. IF absent and it would have had a value other than `success`
+processing will stop with an informative stop code. Allowed error code
+values for this `status` are:
+
+* `success` - no problem found
+
+* `alloc_fault` - `self` was of type `bitset_large` and allocation of
+  memory failed.
+
+* `array_size_invalid_error` - if the number of bits read from `unit`
+  is either negative or greater than 64, if class of `self` is
+  `bitset_64`.
+
+* `read_failure` - failure during a read statement
+
+#### Example
+
+```fortran
+    program demo_input
+        character(*), parameter :: &
+            bits_0   = '000000000000000000000000000000000', &
+            bits_1   = '000000000000000000000000000000001', &
+            bits_33  = '100000000000000000000000000000000'
+        integer :: unit
+        type(bitset_64) :: set0, set1, set2, set3, set4, set5
+        call set0 % from_string( bits_0 )
+        call set1 % from_string( bits_1 )
+        call set2 % from_string( bits_33 )
+        open( newunit=unit, file='test.bin', status='replace', &
+            form='unformatted', action='write' )
+        call set2 % output(unit)
```

Comment:
Maybe I missed it, but would it be possible to use:
```fortran
 call set2 % output('test.bin')
```
and similarly for `input`?